src/reader.rs

```diff
@@ -0,0 +1,71 @@
+use crate::{AsepriteParseError, Result};
+use byteorder::{LittleEndian, ReadBytesExt};
+use flate2::read::ZlibDecoder;
+use std::io::{Cursor, Read, Seek, SeekFrom};
+
+pub(crate) struct AseReader<T: Read + Seek> {
```

Comment:
This new struct `AseReader` wraps over some common reading behaviors:

- Bakes in `<LittleEndian>` so we don't have to specify it everywhere
- Uses methods named after the terms in the Aseprite file spec (WORD, DWORD, etc)
- Exposes some methods that were previously done in helper functions like reading strings, skipping bytes, and filling a buffer.

This change requires the underlying type to be both `Read + Seek` to support the skip function. If that's a problem, we can change `skip` to a Read only implementation.

Reply:
Yes, makes sense. It's a nice cleanup.

In the current usage it doesn't even need to be generic. It could just wrap a `Cursor`, because every section is read into a `Vec<u8>` first. So we can't have a fully streaming implementation, but I'm not planning to add this until someone has a very compelling use case.

One thing I'm thinking about is to keep image data compressed in memory, as that might potentially save a lot of RAM. But that's a more complicated task and would have to be a separate PR.